tests/mainnet/metaregistry/integrative/test_getters.py

```diff
@@ -0,0 +1,793 @@
+    all_combinations = base_combinations
+    if is_meta:
+        underlying_coins = [
+            coin for coin in metaregistry.get_underlying_coins(pool) if coin != brownie.ZERO_ADDRESS
+        ]
+        all_combinations = all_combinations + [(pool_coins[0], coin) for coin in underlying_coins]
+
+    for combination in all_combinations:
+        if combination[0] == combination[1]:
+            continue
+        metaregistry_output = metaregistry.get_coin_indices(pool, combination[0], combination[1])
+        actual_output = list(registry.get_coin_indices(pool, combination[0], combination[1]))
+
+        if registry_id == 1:
+            # fix bug with stable factory where returned `is_underlying` is always True
+
+            # so check if pool is metapool and basepool lp token is not in combination
+            # (then `is_underlying` == True)
+            actual_output[-1] = False
+            if registry.is_meta(pool) and not registry.get_coins(pool)[1] in combination:
+                actual_output[-1] = True
+
+        elif registry_id in [2, 3]:
+            # mainnet crypto registry and crypto factory do not return `is_underlying`
+            # but metaregistry does (since stable registry and factory do)
+            actual_output = (actual_output[0], actual_output[1], False)
+
+        assert actual_output == metaregistry_output
+
+
+@pytest.mark.parametrize("pool_id", range(MAX_POOLS))
+def test_get_pool_params_stableswap_cryptoswap(metaregistry, registry_pool_index_iterator, pool_id):
+
+    skip_if_pool_id_gt_max_pools_in_registry(pool_id, registry_pool_index_iterator)
+
+    registry_id, registry_handler, registry, pool = registry_pool_index_iterator[pool_id]
+
+    """This test is only for stableswap and cryptoswap amms"""
+
+    print(f"testing get_pool_params for pool: {pool}")
+    metaregistry_output = metaregistry.get_pool_params(pool)
+    print(f"metaregistry output: {metaregistry_output}")
+    actual_pool_params = [0] * 20
+
+    v2_pool = curve_pool_v2(pool)
+
+    # A
+    if registry_id != METAREGISTRY_CRYPTO_FACTORY_HANDLER_INDEX:
+        actual_pool_params[0] = registry.get_A(pool)
+    else:
+        actual_pool_params[0] = v2_pool.A()
+
+    # D, gamma
+    if registry_id == METAREGISTRY_CRYPTO_REGISTRY_HANDLER_INDEX:
+        actual_pool_params[1] = registry.get_D(pool)
+        actual_pool_params[2] = registry.get_gamma(pool)
+    elif registry_id == METAREGISTRY_CRYPTO_FACTORY_HANDLER_INDEX:
+        actual_pool_params[1] = v2_pool.D()
+        actual_pool_params[2] = v2_pool.gamma()
+
+    # allowed_extra_profit
+    if registry_id in [
+        METAREGISTRY_CRYPTO_REGISTRY_HANDLER_INDEX,
+        METAREGISTRY_CRYPTO_FACTORY_HANDLER_INDEX,
+    ]:
+        actual_pool_params[3] = v2_pool.allowed_extra_profit()
+        actual_pool_params[4] = v2_pool.fee_gamma()
+        actual_pool_params[5] = v2_pool.adjustment_step()
+        actual_pool_params[6] = v2_pool.ma_half_time()
```

Comment:
could write this as one-liner (`actual_pool_params[3:7] = [v2_pool.allowed_extra_profit(), v2_pool.fee_gamma()...`)